src/main/java/io/fabric8/maven/docker/WatchMojo.java

```diff
@@ -176,6 +176,9 @@ private Runnable createBuildWatchTask(final ServiceHub hub, final ImageWatcher w
             throws MojoExecutionException {
         final ImageConfiguration imageConfig = watcher.getImageConfiguration();
         final AssemblyFiles files = hub.getArchiveService().getAssemblyFiles(imageConfig, mojoParameters);
+        if (files.isEmpty()) {
+            log.error("No assembly files for " + imageConfig + ". Are you sure you invoked this goal after the `package` goal?");
+        }
```

Comment:
Problem would be that this change is not backwards compatible for those already binding `docker:build` to a phase. So I probably postpone this until the 0.16.x release and just leave the error here.

Do you think we should even throw an exception here because its an error ? If its only a warning, then maybe we should also use only a warning message.